src/decoder/index.ts

```diff
@@ -0,0 +1,101 @@
+import { methodID, rawDecode } from 'ethereumjs-abi';
+import { bufferToHex } from 'ethereumjs-util';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { formatValue } from './types';
+import { Contract } from '../contract';
+import { ContractABI, MethodABI, Parameter } from '../contract/json';
+import { ensure } from '../util/ensure';
+import { isValidJSON } from '../util/json';
+
+/**
+ * A class to decode contract calls and explain their purpose
+ */
+export class Decoder {
+
+  /**
+     * Read in and parse all methods from all defined contract abis
```

Comment:
Nit: Spacing

Reply:
the linter did that :laughing:

Reply:
I can't find the rule for it to disable it right now, but the nice thing about auto-fixing linters is that we don't have to worry about style nits like this and can just update the linter and re-fix if we decide later

Reply:
Ah - found the linter rule for this. fixed ;)